src/diagnostics/MOM_spatial_means.F90

```diff
@@ -330,56 +363,65 @@ function global_mass_integral(h, G, GV, var, on_PE_only, scale, tmp_scale, unsca
-  real,          optional, intent(in)  :: tmp_scale !< A temporary rescaling factor for the
-                                               !! variable that is reversed in the return value [a A-1 ~> 1]
+  real,          optional, intent(in)  :: tmp_scale !< A temporary rescaling factor for the variable
+                                               !! that is reversed in the return value [a A-1 ~> 1],
+                                               !! or [b B-1 ~> 1] if unscale is also present.
   real,          optional, intent(in)  :: unscale !< A rescaling factor for the variable [a A-1 ~> 1]
                                                !! that converts it back to unscaled (e.g., mks)
-                                               !! units to enable the use of the reproducing sums.
+                                               !! units to enable the use of the reproducing sums, or
+                                               !! a factor converting between rescaled units if
+                                               !! tmp_scale is also present [B A-1 ~> b a-1].
                                                !! Here scale and unscale are synonymous, but unscale
                                                !! is preferred and takes precedence if both are present.
-  real :: global_mass_integral  !< The mass-weighted integral of var (or 1) in
-                                !! kg times the arbitrary units of var [kg a] or [kg A ~> kg a]
+  real :: global_mass_integral  !< The mass-weighted integral of var (or 1) in kg times the arbitrary
+                                !! units of var [kg a] or in [R Z L2 A ~> kg a] if tmp_scale is present
+                                !! or [R Z L2 B ~> kg b] if both unscale and tmp_scale are present
 
   ! Local variables
-  ! In the following comments, [A] is used to indicate the arbitrary, possibly rescaled units of the
+  ! In the following comments, [A ~> a] is used to indicate the arbitrary, possibly rescaled units of the
   ! input array while [a] indicates the unscaled (e.g., mks) units that can be used with the reproducing sums
-  real, dimension(SZI_(G),SZJ_(G)) :: tmpForSumming ! The mass-weighted integral of the variable in a column [kg a]
-  real :: scalefac   ! An overall scaling factor for the cell mass and variable [a kg A-1 H-1 L-2 ~> kg m-3 or 1]
-  real :: temp_scale ! A temporary scaling factor [1] or [a A-1 ~> 1]
+  ! [A ~> a] and [B ~> b] are the same units unless tmp_scale and unscale are both present.
+  real, dimension(SZI_(G),SZJ_(G)) :: tmpForSumming ! The mass-weighted integral of the variable in a
+                     ! column [kg a] or [kg] or [B R Z L2 ~> kg b] or [R Z L2 ~> kg] if tmp_scale is present.
+  real :: scalefac   ! An overall scaling factor for the cell mass and variable [a kg A-1 R-1 Z-1 L-2 ~> 1]
+                     ! or [1] or [B A-1 ~> b a-1] if tmp_scale is present.
```

Comment:
The description of `scalefac` is missing one possible unit (kg R-1 Z-1 L-2).

|`tmp_scale` | `unscale` | `scalefac` | `tmpForSumming` |
|-|-|-|-|
|T |T |[B A-1 ~> b a-1] |[B R Z L2 ~>  kg b] |
|T |F |[1] |[R Z L2 ~> kg] |
|F |T |[a kg A-1 R-1 Z-1 L-2 ~> 1] |[kg a] |
|F |F |[kg R-1 Z-1 L-2 ~>1] |[kg] |

Reply:
I have added  a version of this table in the comments.